doc/TUTORIAL.md

```diff
@@ -54,177 +46,197 @@ This information allows other humans to help and build on your work.
+
+This code allows us to call this URL <https://img.shields.io/gem/v/formatador.svg> to generate this badge: ![](https://img.shields.io/gem/v/formatador.svg)
+
+It is also worth considering the code we _haven't_ written here. Note that our example doesn't contain any explicit error handling code, but our badge handles errors gracefully. For example, if we call https://img.shields.io/gem/v/does-not-exist.svg we render a 'not found' badge ![](https://img.shields.io/gem/v/does-not-exist.svg) because https://rubygems.org/api/v1/gems/this-package-does-not-exist.json returns a `404 Not Found` status code. When dealing with well-behaved APIs, some of our error handling will be handled implicitly in `BaseJsonService`.
+
+Specifically `BaseJsonService` will handle the following errors for us:
+
+* API does not respond
+* API responds with a non-`200 OK` status code
+* API returns a response which can't be parsed as JSON
+* API returns a response which doesn't validate against our schema
+
+Sometimes it may be necessary to manually throw an exception to deal with a non-standard error condition. If so, standard exceptions can be imported from [errors.js](https://github.com/badges/shields/blob/master/services/errors.js) and thrown.
+
+### (4.4) Adding an Example to the Front Page
+
+Once we have implemented our badge, we can add it to the index so that users can discover it. We will do this by adding a couple of additional methods to our class.
+
 
 ```js
-{
-  title: 'Test Badge from the tutorial',
-  keywords: [
-    'some-search-keyword'
-  ],
-  previewUrl: '/test/subject/STATUS.svg',
-},
+module.exports = class GemVersion extends BaseJsonService {
+
+  // ...
+
+  static get category() {                                       // (1)
+    return 'version'
+  }
+
+  static get examples() {                                       // (2)
+    return [
+      {                                                         // (3)
+        title: 'Gem',
+        urlPattern: ':package',
+        staticExample: this.render({ version: '2.1.0' }),
+        exampleUrl: 'formatador',
+        keywords: ['ruby'],
+      },
+    ]
+  }
+
+}
 ```
 
-Save, run `npm run build`, and you can see it [locally][home].
+1. The `category()` property defines which heading in the index our example will appear under.
+2. The examples property defines an array of examples. In this case the array will contain a single object, but in some cases it is helpful to provide multiple usage examples.
+3. Our example object should contain the following properties:
+    * `title`: Descriptive text that will be shown next to the badge
+    * `urlPattern`: Describe the variable part of the URL using `:param` syntax.
```

Comment:
Tangent:

For the moment, lets just document this but I wonder if we could actually find a way to generate this from the url regex and capture vars instead of requiring the user to manually define it. I am reminded of this conversation https://github.com/badges/shields/pull/1582#discussion_r176912366

Reply:
Yea, I think the solution may be to do this in the opposite direction: we define the `urlPattern` and automatically generate the regex using [url-pattern](https://github.com/snd/url-pattern).